app/src/main/java/com/amaze/filemanager/activities/MainActivity.java

```diff
@@ -487,65 +486,74 @@ private void checkForExternalIntent(Intent intent) {
-                mReturnIntent = true;
-                mRingtonePickerIntent = true;
-                Toast.makeText(this, getString(R.string.pick_a_file), Toast.LENGTH_LONG).show();
-
-                // disable screen rotation just for convenience purpose
-                // TODO: Support screen rotation when picking file
-                Utils.disableScreenRotation(this);
-            } else if (actionIntent.equals(Intent.ACTION_VIEW)) {
-                // zip viewer intent
-                Uri uri = intent.getData();
-
-                if (type != null && type.equals(ARGS_INTENT_ACTION_VIEW_MIME_FOLDER)) {
-                    // support for syncting or intents from external apps that
-                    // need to start file manager from a specific path
-
-                    if (uri != null) {
-
-                        path = Utils.sanitizeInput(uri.getPath());
-                    } else {
-                        // no data field, open home for the tab in later processing
-                        path = null;
-                    }
+        if (actionIntent == null) {
+            return;
+        }
+
+        if (actionIntent.equals(Intent.ACTION_GET_CONTENT)) {
+            // file picker intent
+            mReturnIntent = true;
+            Toast.makeText(this, getString(R.string.pick_a_file), Toast.LENGTH_LONG).show();
+
+            // disable screen rotation just for convenience purpose
+            // TODO: Support screen rotation when picking file
+            Utils.disableScreenRotation(this);
+        } else if (actionIntent.equals(RingtoneManager.ACTION_RINGTONE_PICKER)) {
+            // ringtone picker intent
+            mReturnIntent = true;
+            mRingtonePickerIntent = true;
+            Toast.makeText(this, getString(R.string.pick_a_file), Toast.LENGTH_LONG).show();
+
+            // disable screen rotation just for convenience purpose
+            // TODO: Support screen rotation when picking file
+            Utils.disableScreenRotation(this);
+        } else if (actionIntent.equals(Intent.ACTION_VIEW)) {
+            // zip viewer intent
+            Uri uri = intent.getData();
+
+            if (type != null && type.equals(ARGS_INTENT_ACTION_VIEW_MIME_FOLDER)) {
+                // support for syncting or intents from external apps that
+                // need to start file manager from a specific path
+
+                if (uri != null) {
+
+                    path = Utils.sanitizeInput(uri.getPath());
                 } else {
-                    // we don't have folder resource mime type set, supposed to be zip/rar
-                    openzip = true;
-                    zippath = Utils.sanitizeInput(uri.toString());
+                    // no data field, open home for the tab in later processing
+                    path = null;
+                }
+            } else if (CompressedMimeParsing.isFileCompressed(type) && uri != null) {
+                if(uri.toString().startsWith("content")) {
```

Comment:
Why?

Reply:
Content uris are not supported for decompression/viewing.

Reply:
Please be reminded that `java.io.File` based file I/O operations should be avoided where possible since Android 10 will require use of SAF instead.

https://android.jlelse.eu/handling-files-in-code-after-the-android-10-released-2bea0e16d35

Reply:
Should this be marked resolved?